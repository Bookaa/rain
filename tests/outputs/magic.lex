keyword 'let'
name 'wrapper'
symbol '='
'table'
newline
name 'wrapper'
symbol '.'
name 'new'
symbol '='
keyword 'func'
symbol '('
name 'self'
symbol ','
name 'value'
symbol ')'
indent
keyword 'return'
symbol '{'
name 'value'
symbol '='
name 'value'
symbol '}'
operator '::'
name 'self'
newline
dedent
newline
name 'wrapper'
symbol '['
string '_str'
symbol ']'
symbol '='
keyword 'func'
symbol '('
name 'self'
symbol ')'
indent
keyword 'return'
string 'wrap('
operator '$'
name 'tostr'
symbol '('
name 'self'
symbol '.'
name 'value'
symbol ')'
operator '$'
string ')'
newline
dedent
newline
name 'wrapper'
symbol '['
string '_add'
symbol ']'
symbol '='
keyword 'func'
symbol '('
name 'self'
symbol ','
name 'other'
symbol ')'
indent
keyword 'return'
symbol '{'
name 'value'
symbol '='
name 'self'
symbol '.'
name 'value'
operator '+'
name 'other'
symbol '.'
name 'value'
symbol '}'
operator '::'
name 'meta'
symbol '('
name 'self'
symbol ')'
newline
dedent
newline
name 'wrapper'
symbol '['
string '_mul'
symbol ']'
symbol '='
keyword 'func'
symbol '('
name 'self'
symbol ','
name 'other'
symbol ')'
indent
keyword 'return'
symbol '{'
name 'value'
symbol '='
name 'self'
symbol '.'
name 'value'
operator '*'
name 'other'
symbol '.'
name 'value'
symbol '}'
operator '::'
name 'meta'
symbol '('
name 'self'
symbol ')'
newline
dedent
newline
name 'wrapper'
symbol '['
string '_sub'
symbol ']'
symbol '='
keyword 'func'
symbol '('
name 'self'
symbol ','
name 'other'
symbol ')'
indent
keyword 'return'
symbol '{'
name 'value'
symbol '='
name 'self'
symbol '.'
name 'value'
operator '-'
name 'other'
symbol '.'
name 'value'
symbol '}'
operator '::'
name 'meta'
symbol '('
name 'self'
symbol ')'
newline
dedent
newline
name 'wrapper'
symbol '['
string '_div'
symbol ']'
symbol '='
keyword 'func'
symbol '('
name 'self'
symbol ','
name 'other'
symbol ')'
indent
keyword 'return'
symbol '{'
name 'value'
symbol '='
name 'self'
symbol '.'
name 'value'
operator '/'
name 'other'
symbol '.'
name 'value'
symbol '}'
operator '::'
name 'meta'
symbol '('
name 'self'
symbol ')'
newline
dedent
newline
keyword 'let'
name 'main'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
keyword 'let'
name 'a'
symbol '='
name 'wrapper'
symbol ':'
name 'new'
symbol '('
int 40
symbol ')'
newline
keyword 'let'
name 'b'
symbol '='
name 'wrapper'
symbol ':'
name 'new'
symbol '('
int 8
symbol ')'
newline
keyword 'let'
name 'add'
symbol '='
name 'a'
operator '+'
name 'b'
newline
keyword 'let'
name 'mul'
symbol '='
name 'a'
operator '*'
name 'b'
newline
keyword 'let'
name 'div'
symbol '='
name 'a'
operator '/'
name 'b'
newline
keyword 'let'
name 'sub'
symbol '='
name 'a'
operator '-'
name 'b'
newline
name 'print'
symbol '('
name 'a'
symbol ')'
newline
name 'print'
symbol '('
name 'b'
symbol ')'
newline
name 'print'
symbol '('
name 'add'
symbol ')'
newline
name 'print'
symbol '('
name 'mul'
symbol ')'
newline
name 'print'
symbol '('
name 'div'
symbol ')'
newline
name 'print'
symbol '('
name 'sub'
symbol ')'
newline
dedent
newline
EOF
