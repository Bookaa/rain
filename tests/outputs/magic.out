wrap(40)
wrap(8)
wrap(48)
wrap(320)
wrap(5)
wrap(32)
